### PR TITLE
fix warnings

### LINKED ISSUE #26: msvc 2015 warnings
The code below doesn't compile with the newest winapi modules and /WX

`
boost::process::child c("c:/windows/system32/foo.exe", "/r", dll);
c.wait();
`

Visual Studio 2015
64-bit debug build
boost 1.62.0 static build, with static runtime
flags: /WX /W4 
pp: _SCL_SECURE_NO_WARNINGS;_CRT_SECURE_NO_WARNINGS;

`
Severity	Code	Description	Project	File	Line	Suppression State
Warning	C4244	'argument': conversion from '__int64' to 'int', possible loss of data	fpreg	D:\src\trunk\third_party\boost\include\boost\process\detail\windows\locale.hpp	49	
Warning	C4244	'argument': conversion from '__int64' to 'int', possible loss of data	fpreg	D:\src\trunk\third_party\boost\include\boost\process\detail\windows\locale.hpp	71	
Warning	C4800	'boost::detail::winapi::BOOL_': forcing value to bool 'true' or 'false' (performance warning)	fpreg	D:\src\trunk\third_party\boost\include\boost\process\detail\windows\child_handle.hpp	85	
Warning	C4800	'boost::detail::winapi::BOOL_': forcing value to bool 'true' or 'false' (performance warning)	fpreg	D:\src\trunk\third_party\boost\include\boost\process\detail\windows\child_handle.hpp	92	
Warning	C4800	'boost::detail::winapi::BOOL_': forcing value to bool 'true' or 'false' (performance warning)	fpreg	D:\src\trunk\third_party\boost\include\boost\process\detail\windows\group_handle.hpp	146	
Warning	C4800	'boost::detail::winapi::BOOL_': forcing value to bool 'true' or 'false' (performance warning)	fpreg	D:\src\trunk\third_party\boost\include\boost\process\detail\windows\group_handle.hpp	153	
Warning	C4800	'boost::detail::winapi::BOOL_': forcing value to bool 'true' or 'false' (performance warning)	fpreg	D:\src\trunk\third_party\boost\include\boost\process\detail\windows\group_handle.hpp	183	
Warning	C4099	'boost::process::detail::windows::group_handle': type name first seen using 'struct' now seen using 'class'	fpreg	D:\src\trunk\third_party\boost\include\boost\process\detail\windows\wait_group.hpp	16	
Warning	C4100	'state': unreferenced formal parameter	fpreg	D:\src\trunk\third_party\boost\include\boost\process\detail\windows\locale.hpp	38	
Warning	C4100	'state': unreferenced formal parameter	fpreg	D:\src\trunk\third_party\boost\include\boost\process\detail\windows\locale.hpp	60	
d:\src\trunk\third_party\boost\include\boost\fusion\container\generation\detail\preprocessed\make_set10.hpp	35	
Warning	C4459	declaration of 'env' hides global declaration	fpreg	D:\src\trunk\third_party\boost\include\boost\process\environment.hpp	57	
Warning	C4458	declaration of 'ios' hides class member	fpreg	D:\src\trunk\third_party\boost\include\boost\process\async.hpp	59	
Warning	C4459	declaration of 'on_error' hides global declaration	fpreg	D:\src\trunk\third_party\boost\include\boost\process\detail\windows\executor.hpp	183	
Warning	C4459	declaration of 'on_error' hides global declaration	fpreg	D:\src\trunk\third_party\boost\include\boost\process\detail\windows\executor.hpp	215	
Warning	C4459	declaration of 'on_setup' hides global declaration	fpreg	D:\src\trunk\third_party\boost\include\boost\process\detail\windows\executor.hpp	178	
Warning	C4459	declaration of 'on_success' hides global declaration	fpreg	D:\src\trunk\third_party\boost\include\boost\process\detail\windows\executor.hpp	206

### PR DESCRIPTION
**Pull request by frederich**
fixes #26

### CODE REVIEW

**Comment by frederich** [comment]
Dear Clements,

here some warning fixes. My application is warning free now, but some boost.process unit-test still have warnings.

Jens

File: include/boost/process/detail/windows/child_handle.hpp [hunk 1]
```diff
@@ -82,14 +82,14 @@ struct child_handle
         ::boost::detail::winapi::BOOL_ value;
         if (!::boost::detail::winapi::IsProcessInJob(proc_info.hProcess, nullptr, &value))
             throw_last_error("IsProcessinJob Failed");
-        return value;
+        return value==1; // TODO: correct Windows API usage is: value==TRUE but TRUE isn't defined at winapi
```
Comment (klemens-morgenstern) [inline include/boost/process/detail/windows/child_handle.hpp]: That's not entirely correct. Afaik the guarantee you have is that `FALSE == 0` and `TRUE != 0`, which is also in the [documentation](https://msdn.microsoft.com/de-de/library/windows/desktop/ms684127.aspx) of the function, where it states  "If the function succeeds, the return value is nonzero."

Hence it should be `value != 0`.

Comment (frederich) [inline include/boost/process/detail/windows/child_handle.hpp]: Yeah, you're right, I'll fix it.

File: include/boost/process/detail/windows/locale.hpp [hunk 2]
```diff
@@ -39,14 +40,15 @@ class windows_file_codecvt
      const char* from, const char* from_end, const char*& from_next,
      wchar_t* to, wchar_t* to_end, wchar_t*& to_next) const override
    {
+     boost::ignore_unused(state);
```
Comment (frederich) [inline include/boost/process/detail/windows/locale.hpp]: Better style or not, that's not the point, the point is to fix the warning. Remove the parameter or mark it as unused.

**Comment by klemens-morgenstern** [comment]
Dear Jens,

I'd be really thankful if you would manage to spell my name correctly, especially if it's written all over the screen since it's my github page. Shouldn't be too hard.

I've commented on your changes, if you address them, I'll merge it. Thanks,

Klemens

File: test/Jamfile.jam [hunk 3]
```diff
@@ -16,8 +16,11 @@ if [ os.name ] = NT
 }
 
 project : requirements
-  <toolset>msvc:<define>_SCL_5SECURE_NO_WARNINGS
+  <toolset>msvc:<define>_SCL_SECURE_NO_WARNINGS
+  <toolset>msvc:<define>_CRT_SECURE_NO_DEPRECATE
   <target-os>windows:<define>WIN32_LEAN_AND_MEAN
+  <target-os>windows:<define>NOMINMAX
+  <target-os>windows:<define>_WIN32_WINNT=0x0501
```
Comment (frederich) [inline test/Jamfile.jam]: Why windows version in the 'test' Jamfile', that's why:

..skipped <p..\..\..\bin.v2\libs\process\test\msvc-14.0\debug\threading-multi>sparring_partner.exe for lack of <p..\..\..\bin.v2\libs\filesystem\build\msvc-14.0\debug\threading-multi>boost_filesystem-vc140-mt-gd-1_62.lib...
...skipped <p..\..\..\bin.v2\libs\process\test\msvc-14.0\debug\threading-multi>sparring_partner.pdb for lack of <p..\..\..\bin.v2\libs\filesystem\build\msvc-14.0\debug\threading-multi>boost_filesystem-vc140-mt-gd-1_62.lib...
compile-c-c++ ..\..\..\bin.v2\libs\process\test\msvc-14.0\debug\threading-multi\sub_launcher.obj
sub_launcher.cpp
Please define _WIN32_WINNT or _WIN32_WINDOWS appropriately. For example:
- add -D_WIN32_WINNT=0x0501 to the compiler command line; or
- add _WIN32_WINNT=0x0501 to your project's Preprocessor Definitions.
Assuming _WIN32_WINNT=0x0501 (i.e. Windows XP target).
...skipped <p..\..\..\bin.v2\libs\process\test\msvc-14.0\debug\threading-multi>sub_launch.exe for lack of <p..\..\..\bin.v2\libs\filesystem\build\msvc-14.0\debug\threading-multi>boost_filesystem-vc140-mt-gd-1_62.lib...
...skipped <p..\..\..\bin.v2\libs\process\test\msvc-14.0\debug\threading-multi>sub_launch.pdb for lack of <p..\..\..\bin.v2\libs\filesystem\build\msvc-14.0\debug\threading-multi>boost_filesystem-vc140-mt-gd-1_62.lib...
compile-c-c++ ..\..\..\bin.v2\libs\process\test\environment.test\msvc-14.0\debug\threading-multi\environment.obj

**Comment by frederich** [comment]
Please keep cool, sorry Klemens.

**Pull request merged**